sed/loader/flash/loader.py

```diff
@@ -0,0 +1,589 @@
+"""
+This module implements the flash data loader.
+The raw hdf5 data is saved into parquet files and loaded as a dask dataframe.
+If there are multiple files, the NaNs are forward filled.
+"""
+import os
+from functools import reduce
+from itertools import compress
+from pathlib import Path
+from typing import cast
+from typing import List
+from typing import Sequence
+from typing import Tuple
+from typing import Union
+
+import dask.dataframe as dd
+import h5py
+import numpy as np
+from joblib import delayed
+from joblib import Parallel
+from pandas import DataFrame
+from pandas import MultiIndex
+from pandas import Series
+
+from sed.loader.base.loader import BaseLoader
+from sed.loader.exception_handlers import NoFilesFoundError
+from sed.loader.utils import gather_flash_files
+from sed.loader.utils import parse_h5_keys
+
+
+class FlashLoader(BaseLoader):
+    """
+    The class generates multiindexed multidimensional pandas dataframes
+    from the new FLASH dataformat resolved by both macro and microbunches
+    alongside electrons.
+    """
+
+    __name__ = "flash"
+
+    supported_file_types = ["h5", "parquet"]
+
+    def __init__(self, config: dict, meta_handler) -> None:
+
+        super().__init__(config=config, meta_handler=meta_handler)
+        self._config: dict = config.get("dataframe", {})
+        self.config_parser()
```

Comment:
I find this somewhat misleading here. It defines attributes, not sure if this is a clean way to do it. Why not keep everything inside the _config dict?

Reply:
The attributes are created because they are used in different methods of the loader. Would you suggest I keep calling _config everytime?